### PR TITLE
Object-oriented mailroom

### PR DESCRIPTION
**Pull request by theseanhodges**

### CODE REVIEW

File: students/shodges/lesson09/cli_main.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+
+from donor_models import *
```
Comment (natasha-aleksandrova) [inline students/shodges/lesson09/cli_main.py]: avoid using star imports - it's a bad practice

File: students/shodges/lesson09/donor_models.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+    @property
+    def count(self):
+        """
+        Returns the number of donations in the donor's ledger.
+        """
+        return len(self.ledger)
+
+
+    @property
```
Comment (natasha-aleksandrova) [inline students/shodges/lesson09/donor_models.py]: class methods only need one line space in between

File: students/shodges/lesson09/donor_models.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        return sum(self.ledger)
+
+
+    def process(self, donation):
+        """
+        Appends donation to the donor's ledger.  Returns True if successful.  No exceptions
+        raised are caught and are required to be caught by the implementing code.
+        """
+        self.ledger.append(float(donation))
+        return True
```
Comment (natasha-aleksandrova) [inline students/shodges/lesson09/donor_models.py]: why return True? not needed here

File: students/shodges/lesson09/donor_models.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+    @property
+    def donors(self):
+        """
+        Returns a list of all donors in the database.
+        """
+        donorlist = []
+        for k in self.db.keys():
+            donorlist.append(k)
+        return donorlist
```
Comment (natasha-aleksandrova) [inline students/shodges/lesson09/donor_models.py]: can you use a comprehension here 😉

File: students/shodges/lesson09/donor_models.py (hunk 3 above)
Comment (theseanhodges) [reply students/shodges/lesson09/donor_models.py]: You're right.  For some things I had been checking the return but not here (although I did in a few unit tests so those are fixed).

File: students/shodges/lesson09/donor_models.py (hunk 4 above)
Comment (theseanhodges) [reply students/shodges/lesson09/donor_models.py]: Done!  Submitted changes in #450

**Pull request merged**